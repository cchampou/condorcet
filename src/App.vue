<template>
  <div>
    <Navbar v-if="isAuth"></Navbar>

    <div class="container mx-auto">
      <Notifications :notifications="notifications" />

      <transition name="fade" mode="out-in">
        <router-view></router-view>
      </transition>
    </div>
  </div>
</template>

<script>
import { mapState, mapGetters } from "vuex";
import Navbar from "./components/Navbar";
import Notifications from "./components/Notifications";

export default {
  name: "App",
  components: {
    Navbar,
    Notifications
  },
  computed: {
    ...mapState("app", ["notifications"]),
    ...mapGetters("user", ["isAuth"]),
    canGoBack() {
      return this.$route.path !== "/polls";
    }
  }
};
</script>

<style>
.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.2s;
}
.fade-enter, .fade-leave-to /* .fade-leave-active below version 2.1.8 */ {
  opacity: 0;
}
</style>
