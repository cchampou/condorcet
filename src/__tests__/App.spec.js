import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";

import App from "../App";

const localVue = createLocalVue();
localVue.use(Vuex);

const $route = {
  path: "/polls",
  params: "adz93dze823j"
};

describe("App.vue", () => {
  let store;
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        user: {
          namespaced: true,
          getters: {
            isAuth: () => true
          }
        },
        app: {
          namespaced: true,
          state: { notifications: [] }
        }
      }
    });
  });
  it("should render", () => {
    const wrapper = shallowMount(App, {
      localVue,
      store,
      mocks: {
        $route
      },
      stubs: ["router-link", "router-view"]
    });

    expect(wrapper).toMatchSnapshot();
  });
});
