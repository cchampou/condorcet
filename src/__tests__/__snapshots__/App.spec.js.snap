// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render 1`] = `
<div>
  <navbar-stub></navbar-stub>
  <div class="container mx-auto">
    <notifications-stub notifications=""></notifications-stub>
    <router-view-stub name="fade" mode="out-in"></router-view-stub>
  </div>
</div>
`;
